components/constellation/constellation.rs

```diff
@@ -3097,11 +3117,13 @@ where
             change.browsing_context_id, change.new_pipeline_id
         );
 
-        // If the currently focused pipeline is the one being changed (or a child
-        // of the pipeline being changed) then update the focus pipeline to be
-        // the replacement.
-        if self.focused_pipeline_is_descendant_of(change.browsing_context_id) {
-            self.focus_pipeline_id = Some(change.new_pipeline_id);
+        // If the currently focused browsing context is a child of the browsing
+        // context in which the page is being loaded, then update the focused
+        // browsing context to that one.
+        //
+        // TODO(mandreyel): is this correct?
```

Comment:
This looks correct, but like you said this doesn't really give us the benefits mentioned from the issue. A simple way to not have to iterate the frame tree each time is to check if the currently focused browsing context is the one that is being changed. I believe this will be the most common case (just navigating between pages without interacting with iframes). The top-level check would also work, but I am not sure if that gives us much benefit over the approach I just mentioned.